kitsune/users/handlers.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import Any
+
+from kitsune.sumo.redis_utils import redis_client
+from kitsune.users.models import User
+
+
+@dataclass
+class UserDeletionListener(ABC):
+    """
+    Listener that is responsible for deleting a user.
+    """
+
+    @abstractmethod
+    def on_user_deletion(self, user: User):
+        pass
+
+    def mark_as_processed(self, user: User):
+        redis = UserDeletionPublisher._redis
+        key = f"user_deletion:{user.id}"
+        redis.srem(key, self.__class__.__name__)
+
+
+@dataclass
+class UserDeletionPublisher:
+    """
+    Publisher that is responsible for publishing a user deletion event.
+    """
+
+    user: User
+    listeners: list[UserDeletionListener] = field(default_factory=list)
+    _redis: Any = field(default=None, init=False, repr=False)
+    redis_key: str = field(init=False)
+
+    def __post_init__(self):
+        if UserDeletionPublisher._redis is None:
+            UserDeletionPublisher._redis = redis_client("account_events")
```

Comment:
I don't understand the need for Redis. As far as I can tell, all of the subscribed listeners are run sequentially when `publish()` is called, and none of the listeners that have been defined in this PR do anything asynchronously (like submit their work to a Celery task) within their `on_user_deletion` method. So when the `publish()` completes, it seems to me that all of the work has already been completed. I can't see any need to check `completed_processing()`.